substrate/frame/support/src/lib.rs

```diff
@@ -1106,28 +1106,51 @@ pub mod pallet_macros {
 	/// ```
 	/// #[frame_support::pallet]
 	/// mod pallet {
-	/// # 	use frame_support::pallet_prelude::*;
-	/// # 	use frame_system::pallet_prelude::*;
-	/// #
+	///  	use frame_support::pallet_prelude::*;
+	///  	use frame_system::pallet_prelude::*;
+	///     use sp_weights::Weight;
+	/// 
 	/// 	#[pallet::pallet]
 	/// 	pub struct Pallet<T>(_);
 	///
+	///  	#[pallet::config]
+	///  	pub trait Config: frame_system::Config {
+	///         /// Type for specifying dispatchable weights.
+	///         type WeightInfo: WeightInfo;
+	///     }
+	///
 	/// 	#[pallet::call]
 	/// 	impl<T: Config> Pallet<T> {
-	/// 		#[pallet::weight({0})] // <- set actual weight here
+	/// 		#[pallet::weight(<T as Config>::WeightInfo::do_something())]
 	/// 		#[pallet::call_index(0)]
-	/// 		pub fn something(
-	/// 			_: OriginFor<T>,
+	/// 		pub fn do_something(
+	/// 			origin: OriginFor<T>,
 	/// 			foo: u32,
 	/// 		) -> DispatchResult {
-	/// 			unimplemented!()
+	/// 			Ok(())
 	/// 		}
 	/// 	}
-	/// #
-	/// # 	#[pallet::config]
-	/// # 	pub trait Config: frame_system::Config {}
+	/// }
+	///
+	/// /// The `WeightInfo` trait defines weight functions for dispatchable calls.
+	/// pub trait WeightInfo {
+	///     fn do_something() -> Weight;
 	/// }
 	/// ```
+	/// In this example:
+	///
+	/// - The `WeightInfo` trait defines a function `do_something` that returns the weight for
+	///   the corresponding dispatchable.
+	/// - The `Config` trait includes an associated type `WeightInfo` that implements the
+	///   `WeightInfo` trait.
+	/// - The weight of a dispatchable can also be defined dynamically using the pallet's
+	///   configuration.
+	/// - If the weight is not explicitly defined using the `#[pallet::weight]` attribute, the
```

Comment:
How will it determine the weight?

Reply:
We should include here all the possible syntaxes in the example. To make the users aware.